_posts/2025-11-03-cheriot-1.0.markdown

```diff
@@ -0,0 +1,73 @@
+title:  "CHERIoT 1.0 Released!"
+date:   2025-11-03
+categories: sail specification release
+author: "David Chisnall"
+---
+
+Today, we [released the 1.0 version of the CHERIoT specification](https://github.com/CHERIoT-Platform/cheriot-sail/releases/tag/v1.0)!
+For those reading about CHERIoT for the first time, it is a hardware-software co-design project that aims to produce secure microcontroller-class systems for connected devices.
+We start with a foundational guarantee of memory safety (the hardware will trap on buffer overflows or use after free errors, even in assembly code) and build rich (and useable) compartmentalisation abstractions on top.
+
+This specification defines the ISA, the CHERIoT language extensions, compilation model, relocations, and so on.
+The last change that we made to the ISA was December 2024, so we are confident that this is a stable release that we can support in hardware for a long time.
+This release was implemented by the [1.0 release of CHERIoT Ibex](https://github.com/microsoft/cheriot-ibex/releases/tag/cheriot_ibex_v1.0) and by [CHERIoT Kudu](https://github.com/microsoft/cheriot-kudu) (which has not yet had an official release).
+These two implementations demonstrate that the ISA scales from three-stage single-issue pipelines to six-stage dual-issue pipelines, roughly the same range of microarchitectures supported by Arm's M profile.
+
+We at SCI have the first of our ICENI chips, which use the CHERIoT Ibex core, on the way back from the fab now and will be scaling up to mass production in the new year.
+I am not allowed to speak for other folks building CHERIoT silicon, but I expect 2026 to be an exciting year for the CHERIoT project!
+
+This is a release that, both through the open-source CHERIoT Platform project and through partner companies that aim to ship CHERIoT products, we will be supporting for years to come.
+
+# Alignment with the RISC-V Y base
+
+RISC-V International is currently standardising a CHERI base for RISC-V (tentatively named Y, alongside the existing I and E bases).
+We aim for this to be a common subset across all CHERI implementations, whether they are microcontroller, application cores, or domain-specific accelerators.
+The exact composition of this base is still under discussion, but we aim for CHERIoT 2.0 to be equivalent to CHERIoT 1.0, but built atop this common base.
+
+On the way to 2.0, we will be separating out the parts that are in the common core from those that are CHERIoT-specific.
+We also have the `ct.` prefix reserved for the CHERIoT vendor extensions in RISC-V and so will move any instructions that are not direct equivalents of standardised RISC-V instructions into that namespace.
+For compatibility, out toolchains will support both names, but we expect to transition to the RISC-V official names as the RISC-V standardisation process makes progress.
+
+Expect to see some 1.x releases that make changes in things like assembler mnemonics (but not the underling ISA).
```

Comment:
"underlying"

Reply:
The underling ISA is the you all get to use, I use the overlord ISA.